iXBRLViewerPlugin/viewer/src/js/util.js

```diff
@@ -232,3 +234,17 @@ export function getIXHiddenLinkStyle(domNode) {
     }
     return null;
 }
+
+/**
+ * Moves all attributes from one element to another, excluding data attributes created by the viewer application.
+ */
+export function moveNonAppAttributes(fromElement, toElement) {
+        // Hypen seperated dataset attributes are returned camel case when accessed via jquery .prop().
+        const hypenSepAppAttrPrefix = `data-${IXBRL_VIEWER_DATASET_PREFIX.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase()}`;
```

Comment:
I think it's a DOM `dataset` thing rather than a jQuery thing.

I wonder if copying `.attributes` (but excluding `data-`) and then separately copying `.dataset` would be simpler.

(I wonder if "Designing HTML for Inline XBRL" should recommend a `data-` prefix.)